tests/ha/ha_cluster_init.pm

```diff
@@ -34,8 +45,20 @@ sub run {
     # Note: the default timeout need to be increase because it can takes time to join the cluster
     # Initialize the cluster with diskless or shared storage SBD (default)
     $fencing_opt = '-S' if (get_var('USE_DISKLESS_SBD'));
-    if (script_run "ha-cluster-init -y $fencing_opt $unicast_opt", $join_timeout) {
-        assert_script_run "crm -dR cluster init -y $fencing_opt $unicast_opt", $join_timeout;
+    type_string "ha-cluster-init -y $fencing_opt $unicast_opt $qdevice_opt ; echo ha-cluster-init-finished-\$? > /dev/$serialdev\n";
+    if (get_var('QDEVICE') && (check_screen 'ha-cluster-init_qdevice-password', $join_timeout)) {
+        type_password;
+        send_key 'ret';
+    }
+
+    if (!wait_serial("ha-cluster-init-finished-0", $join_timeout)) {
+        type_string "crm -dR cluster init -y $fencing_opt $unicast_opt $qdevice_opt ; echo ha-cluster-init-finished-\$? > /dev/$serialdev\n";
+        if (get_var('QDEVICE')) {
+            assert_screen 'ha-cluster-init_qdevice-password', $join_timeout;
+            type_password;
+            send_key 'ret';
+        }
+        die "Cluster initialization failed" if (!wait_serial("ha-cluster-init-finished-0", $join_timeout));
     }
```

Comment:
I don't know if using `wait_serial` directly is good, but as we already discussed I don't know a better way for now. Maybe @alvarocarvajald has an idea?

Reply:
Don't see a problem with the use of `wait_serial`. It's already used elsewhere in the HA modules and in several libraries in similar cases (`ha/iscsi_client` comes to mind, for example). I don't think we have a different choice since the password cannot be passed directly as a command line argument.